packages/kamino-sdk/src/Kamino.ts

```diff
@@ -456,41 +454,46 @@ export class Kamino {
-          tokenMintB: new PublicKey(pool.tokenB.mint),
-          tvl: pool.tvl ? new Decimal(pool.tvl) : undefined,
-          feeRate: new Decimal(pool.lpFeeRate),
-          volumeOnLast7d: pool.volume ? new Decimal(pool.volume.week) : undefined,
-          tickSpacing: new Decimal(pool.tickSpacing),
-          // TODO: get real amount of positions
-          positions: new Decimal(0),
-        };
-        return poolInfo;
-      });
+      let genericPoolInfos: GenericPoolInfo[] = await Promise.all(
+        pools.map(async (pool: OrcaPool) => {
+          let positionsCount = new Decimal(await this.getPositionsCountForPool(dex, new PublicKey(pool.address)));
+          let poolInfo: GenericPoolInfo = {
+            dex,
+            address: new PublicKey(pool.address),
+            price: new Decimal(pool.price),
+            tokenMintA: new PublicKey(pool.tokenA.mint),
+            tokenMintB: new PublicKey(pool.tokenB.mint),
+            tvl: pool.tvl ? new Decimal(pool.tvl) : undefined,
+            feeRate: new Decimal(pool.lpFeeRate),
+            volumeOnLast7d: pool.volume ? new Decimal(pool.volume.week) : undefined,
+            tickSpacing: new Decimal(pool.tickSpacing),
+            positions: positionsCount,
+          };
+          return poolInfo;
+        })
+      );
       return genericPoolInfos;
     } else if (dex == 'RAYDIUM') {
       let pools = await this.getRaydiumPoolsForTokens(tokenMintA, tokenMintB);
-      let genericPoolInfos: GenericPoolInfo[] = pools.map((pool: Pool) => {
-        let poolInfo: GenericPoolInfo = {
-          dex,
-          address: new PublicKey(pool.id),
-          price: new Decimal(pool.price),
-          tokenMintA: new PublicKey(pool.mintA),
-          tokenMintB: new PublicKey(pool.mintB),
-          tvl: new Decimal(pool.tvl),
-          feeRate: new Decimal(pool.ammConfig.tradeFeeRate).div(new Decimal(FullBPS)),
-          volumeOnLast7d: new Decimal(pool.week.volume),
-          tickSpacing: new Decimal(pool.ammConfig.tickSpacing),
-          // TODO: get real amount of positions
-          positions: new Decimal(0),
-        };
-        return poolInfo;
-      });
+      let genericPoolInfos: GenericPoolInfo[] = await Promise.all(
+        pools.map(async (pool: Pool) => {
+          let positionsCount = new Decimal(await this.getPositionsCountForPool(dex, new PublicKey(pool.id)));
+          let poolInfo: GenericPoolInfo = {
+            dex,
+            address: new PublicKey(pool.id),
+            price: new Decimal(pool.price),
+            tokenMintA: new PublicKey(pool.mintA),
+            tokenMintB: new PublicKey(pool.mintB),
+            tvl: new Decimal(pool.tvl),
+            feeRate: new Decimal(pool.ammConfig.tradeFeeRate).div(new Decimal(FullBPS)),
+            volumeOnLast7d: new Decimal(pool.week.volume),
+            tickSpacing: new Decimal(pool.ammConfig.tickSpacing),
+            // TODO: get real amount of positions
```

Comment:
remove todo

Reply:
I kept it because I don't have an implementation for Raydium yet and I wanted it to be more obvious but I can remove it

Reply:
okok